boa_engine/src/bytecompiler/mod.rs

```diff
@@ -461,13 +465,46 @@ impl<'ctx, 'host> ByteCompiler<'ctx, 'host> {
     }
 
     pub(crate) fn emit(&mut self, opcode: Opcode, operands: &[Operand]) {
+        let mut varying_kind = VaryingOperandKind::Short;
+        for operand in operands {
+            if let Operand::Varying(operand) = *operand {
+                if u8::try_from(operand).is_ok() {
+                } else if u16::try_from(operand).is_ok() {
+                    varying_kind = VaryingOperandKind::Half;
+                } else {
+                    varying_kind = VaryingOperandKind::Wide;
+                    break;
+                }
+            }
+        }
+
+        match varying_kind {
+            VaryingOperandKind::Short => {}
+            VaryingOperandKind::Half => self.emit_opcode(Opcode::Half),
+            VaryingOperandKind::Wide => self.emit_opcode(Opcode::Wide),
+        }
         self.emit_opcode(opcode);
         for operand in operands {
-            self.emit_operand(*operand);
+            self.emit_operand(*operand, varying_kind);
+        }
+    }
+
+    pub(crate) fn emit_varying_width(&mut self, opcode: Opcode, operand: u32) {
```

Comment:
I think a comment above this function would be useful to explain why we use this over `emit`.